code/modules/projectiles/projectile.dm

```diff
@@ -674,268 +735,412 @@
+		pixels_moved_last_tile += distance_to_move
+		if (!deletion_queued && pixels_moved_last_tile >= ICON_SIZE_ALL)
+			reduce_range()
+			if (QDELETED(src))
+				return
+			// Similarly with range out deletion, need to calculate how many pixels we can actually move before deleting
+			if (deletion_queued)
+				delete_distance = distance_to_move - (pixels_moved_last_tile - ICON_SIZE_ALL)
+
+		if (deletion_queued)
+			// We moved to the next turf first, then impacted something
+			// This means that we need to offset our visual position back to the previous turf, then figure out
+			// how much we moved on the next turf (or we didn't move at all in which case we both shifts are 0 anyways)
+			if (loc == new_turf)
+				pixel_x -= x_shift * ICON_SIZE_X
+				pixel_y -= y_shift * ICON_SIZE_Y
+
+			// Similarly to normal animate code, but use lowered deletion distance instead.
+			var/delete_x = pixel_x + movement_vector.pixel_x * delete_distance
+			var/delete_y = pixel_y + movement_vector.pixel_y * delete_distance
+			// In order to keep a consistent speed, calculate at what point between ticks we get deleted
+			var/animate_time = world.tick_lag * delete_distance / total_move_distance
+			// Sometimes we need to move *just a bit* more than we can afford this tick - in this case, delete a tick after
+			// so we don't disappear before impact. This shouldn't be more than 1, ever.
+			if (delete_distance > pixels_to_move)
+				ticks_to_deletion += 1
+			// We can use animation chains to visually disappear between ticks.
+			if (!move_animate(delete_x, delete_y, animate_time, deleting = TRUE))
+				animate(src, pixel_x = delete_x, pixel_y = delete_y, time = animate_time, flags = ANIMATION_PARALLEL | ANIMATION_CONTINUE)
+				animate(alpha = 0, time = 0, flags = ANIMATION_CONTINUE)
+			return
+
+		pixels_to_move -= distance_to_move
+		// animate() instantly changes pixel_x/y values and just interpolates them client-side so next loop processes properly
+		if (hitscan)
+			pixel_x = entry_x
+			pixel_y = entry_y
+		else
+			// We need to shift back to the tile we were on before moving
+			pixel_x -= x_shift * ICON_SIZE_X
+			pixel_y -= y_shift * ICON_SIZE_Y
+			if (!move_animate(entry_x, entry_y))
+				animate(src, pixel_x = entry_x, pixel_y = entry_y, time = world.tick_lag * distance_to_move / total_move_distance, flags = ANIMATION_PARALLEL | ANIMATION_CONTINUE)
+
+		// Homing caps our movement speed per loop while leaving per tick speed intact, so we can just call process_homing every loop here
+		if (homing)
+			process_homing()
+
+		// We've hit a timestop field, abort any remaining movement
+		if (paused)
+			return
+
+		// Prevents long-range high-speed projectiles from ruining the server performance by moving 100 tiles per tick when subsystem is set to a high cap
+		if (CHECK_TICK)
+			// If we ran out of time, add whatever distance we're yet to pass to overrun debt to be processed next tick and break the loop
+			overrun += pixels_to_move
+			return
```

Comment:
@CHECK_TICK sleeps that doesn't seem like what you'd intend here. you want TICK_CHECK